include/my_bit.h

```diff
@@ -37,6 +37,12 @@ static inline uint my_bit_log2(ulong value)
   return bit;
 }
 
+static inline uint my_bit_size_t_log2(size_t value)
+{
+  uint bit;
+  for (bit=0 ; value > 1 ; value>>=1, bit++) ;
+  return bit;
```

Comment:
would probably be better as type template like https://github.com/MariaDB/server/commit/e7f426d2c9be3ebc7c373e80873996ce8857ab48#diff-28377f2a728b889ab5a14d8e5f31de8eR169 ?